### PR TITLE
NAS-131143 / 25.04 / Always use python serializer and preserve types

### PR DESCRIPTION
**Pull request by themylogin**
…htly different results

### CODE REVIEW

**Comment by anodos325** [comment]
Won't this change the type for `roles` in user.query and `group.query` response?

File: src/middlewared/middlewared/plugins/account.py [hunk 1]
```diff
@@ -1612,11 +1612,13 @@ async def group_extend(self, group, ctx):
         group['name'] = group['group']
         group['users'] = ctx['memberships'].get(group['id'], [])
 
-        privilege_mappings = privileges_group_mapping(ctx['privileges'], [group['gid']], 'local_groups')
+        privilege_mappings = await self.middleware.call(
```
Comment (anodos325) [inline src/middlewared/middlewared/plugins/account.py:1615]: I'd rather not add an extra middleware call per-group here.

**Review by anodos325 — APPROVED** [review]
Looks good.

**Comment by themylogin** [comment]
time 4:00

**Pull request merged**